src/hb-ot-shape-math.cc

```diff
@@ -0,0 +1,717 @@
+  //  \sum_{i extender} [ (FullAdvance_i - minConnectorOverlap)]
+  // and   sumNonExt =
+  //  \sum_{i not extender} [(FullAdvance_i - minConnectorOverlap)]
+  //
+  // that is
+  //
+  // repeatCount =
+  //           ceil [ (target_size - sumNonExt - minConnectorOverlap) / sumExt ]
+
+  unsigned int extCount = 0;
+  hb_position_t sumExt = 0;
+  hb_position_t sumNonExt = 0;
+  for (unsigned int i = 0; i < glyphAssembly.part_count(); i++) {
+    hb_position_t fullAdvanceMinusEndOverlap = glyphAssembly.full_advance(i) - minConnectorOverlap;
+    if (glyphAssembly.is_extender(i)) {
+      sumExt += fullAdvanceMinusEndOverlap;
+      extCount++;
+    } else {
+      sumNonExt += fullAdvanceMinusEndOverlap;
+    }
+  }
+  if (sumExt == 0) return false; // error in the extender count or metrics
+  unsigned int extRepeatCount = 0;
+  if (sumNonExt + minConnectorOverlap < target_size)
+    extRepeatCount =
+      ceil(static_cast<float>(target_size - sumNonExt - minConnectorOverlap) /
+           sumExt);
+
+  // Determine the actual number of glyphs necessary to draw this assembly at
+  // the specified target size.
+  unsigned int glyphCount =
+    (glyphAssembly.part_count() - extCount) + extRepeatCount * extCount;
+  if (glyphCount == 0 ||
+      glyphCount > HB_OT_MATH_MAXIMUM_PART_COUNT_IN_GLYPH_ASSEMBLY)
+    return false;
+  if (!hb_buffer_set_length (buffer, glyphCount)) return false;
+  buffer->content_type = HB_BUFFER_CONTENT_TYPE_GLYPHS;
+  buffer->have_positions = true;
+
+  // Add the glyph to the buffer and position them with the minimal overlap.
+  uint32_t initialCluster = buffer->info[0].cluster;
+  buffer->pos[0].x_offset = buffer->pos[0].y_offset = 0;
+  for (unsigned i = 0, j = 0; i < glyphAssembly.part_count(); i++) {
+    unsigned int partRepeatCount =
+      glyphAssembly.is_extender(i) ? extRepeatCount : 1;
+    for (unsigned int k = 0; k < partRepeatCount; k++, j++) {
+      buffer->info[j].codepoint = glyphAssembly.glyph(i);
+
+      // For now, we set all advances to zero.
+      buffer->pos[j].x_advance = 0;
+      buffer->pos[j].y_advance = 0;
+
+      // We temporarily put each glyph in different cluster, so that we can
+      // easily retrieve the value of i from the value of j below.
+      buffer->info[j].cluster = i;
+
+      if (j < glyphCount) {
+        hb_position_t delta_offset =
+          glyphAssembly.full_advance(i) - minConnectorOverlap;
+        if (glyphAssembly.is_horizontal()) {
+          buffer->pos[j + 1].x_offset = buffer->pos[j].x_offset + delta_offset;
+          buffer->pos[j + 1].y_offset = 0;
+        } else {
+          buffer->pos[j + 1].x_offset = 0;
+          buffer->pos[j + 1].y_offset = buffer->pos[j].y_offset + delta_offset;
+        }
+      }
+    }
+  }
+
+  // Now we try and increase the overlap between parts in order to get closer
+  // to the target size. Again, for symmetry reason we do it by browsing the
+  // from the middle and simultaneously towards the start and end.
```

Comment:
So here we depart a bit from what the MATH table spec says. This preserves the symmetry of the operator but not the condition that things are distributed equally. We should probably do a first round to increase all the overlaps by the same amount. And if that is still not enough maybe do this correction to reduce the size any further.